dali/operators/reader/numpy_reader_op.cc

```diff
@@ -157,19 +304,28 @@ void NumpyReader::RunImpl(HostWorkspace &ws) {
   int ndim = out_sh.sample_dim();
   int nsamples = out_sh.num_samples();
   auto &thread_pool = ws.GetThreadPool();
+  int nthreads = thread_pool.NumThreads();
 
   for (int i = 0; i < nsamples; i++) {
     const auto& file_i = GetSample(i);
     const auto& file_sh = file_i.image.shape();
     bool need_transpose = !(file_i.meta == "transpose:false");
+    bool need_slice = !rois_.empty();
 
     // controls task priority
     int64_t task_sz = volume(file_i.image.shape());
-    if (need_transpose)
+    // slice + transpose is treated as similar size as just slice
+    if (need_slice)
+      task_sz += volume(rois_[i].shape);
+    else if (need_transpose)
       task_sz += out_sh.tensor_size(i);
```

Comment:
Why `+=` ? We don't copy the original data, so it could be a plain assignment.
I'd go for:
```suggestion
    // slice + transpose is treated as similar size as just slice
    if (need_slice)
      task_sz = volume(rois_[i].shape);
    if (need_transpose)
      task_sz *= 2;
```